insalan/tournament/views/group.py

```diff
@@ -0,0 +1,44 @@
+from django.utils.translation import gettext_lazy as _
+from django.core.exceptions import PermissionDenied, BadRequest
+
+from rest_framework import generics, permissions, status
+from rest_framework.response import Response
+from rest_framework.exceptions import NotFound
+
+import insalan.tournament.serializers as serializers
+
+from ..models import Group, GroupMatch, MatchStatus, validate_match_data
+from ..manage import update_match_score
+
+from .permissions import ReadOnly, Patch
+
+from collections import Counter
+
+class GroupMatchScore(generics.UpdateAPIView):
+    """Update score of a group match"""
+
+    queryset = GroupMatch.objects.all().order_by("id")
+    serializer_class = serializers.GroupMatchSerializer
+    permission_classes = [permissions.IsAuthenticated]
+
+    def patch(self, request, *args, **kwargs):
+        user = request.user
+        data = request.data
+
+        try:
+            match = GroupMatch.objects.get(pk=kwargs["match_id"],group=kwargs["group_id"])
```

Comment:
You should probably add a prior check that `"match_id"` and `"group_id"` keys exist *prior* to checking the existence of the object. If these keys do not exist, it should probably be a "bad request", and not a "not found".